### PR TITLE
Convert bntest to TEST_ framework

### PR DESCRIPTION
**Pull request by richsalz**
Converts bntest.  I gave up on #3062 (for now).

### CODE REVIEW

File: test/bntest.c [hunk 1]
```diff
@@ -1617,40 +1545,34 @@ static int test_dec2bn()
...
         goto err;
-    }
     BN_free(bn);
 
-    ret = parsedecBN(&bn, "256");
-    if (ret != 3 || !BN_is_word(bn, 256) || BN_is_negative(bn)) {
-        fprintf(stderr, "BN_dec2bn(256) gave a bad result.\n");
+    if (!TEST_int_eq(parsedecBN(&bn, "256"), 3)
+            || !TEST_true(BN_is_word(bn, 256))
+            || BN_is_negative(bn))
```
Comment (mattcaswell) [inline test/bntest.c]: !TEST_false(BN_is_negative(bn))?

Comment (mattcaswell) [inline test/bntest.c]: Here and several other instances

File: test/bntest.c [hunk 2]
```diff
@@ -1768,57 +1683,34 @@ static const MPITEST kMPITests[] = {
...
-        if (mpi_len != mpi_len2) {
-            fprintf(stderr, "MPI test #%u: length changes.\n",
-                    (unsigned)i);
-            goto err;
-        }
+    if (!TEST_ptr(bn = BN_new())
+            || !TEST_true(BN_asc2bn(&bn, test->base10)))
+        goto err;
+    mpi_len = BN_bn2mpi(bn, NULL);
+    if (!TEST_size_t_lt(mpi_len, sizeof(scratch)))
```
Comment (mattcaswell) [inline test/bntest.c]: TEST_size_t_le?

File: test/bntest.c [hunk 3]
```diff
@@ -1951,78 +1815,57 @@ static int test_negzero()
...
+    if (!TEST_ptr(a = BN_new())
+            || !TEST_ptr(b = BN_new())
+            || !TEST_ptr(zero = BN_new())
+            || !TEST_ptr(mont = BN_MONT_CTX_new()))
         goto err;
     BN_zero(zero);
 
-    if (BN_div(a, b, BN_value_one(), zero, ctx)) {
-        fprintf(stderr, "Division by zero succeeded!\n");
+    if (BN_div(a, b, BN_value_one(), zero, ctx))
```
Comment (mattcaswell) [inline test/bntest.c]: TEST_false?

File: test/bntest.c [hunk 4]
```diff
@@ -1951,78 +1815,57 @@ static int test_negzero()
...
         goto err;
-    }
     ERR_clear_error();
 
-    if (BN_mod_mul(a, BN_value_one(), BN_value_one(), zero, ctx)) {
-        fprintf(stderr, "BN_mod_mul with zero modulus succeeded!\n");
+    if (BN_mod_mul(a, BN_value_one(), BN_value_one(), zero, ctx))
```
Comment (mattcaswell) [inline test/bntest.c]: TEST_false?

File: test/bntest.c [hunk 5]
```diff
@@ -1951,78 +1815,57 @@ static int test_negzero()
...
         goto err;
-    }
     ERR_clear_error();
 
-    if (BN_mod_exp(a, BN_value_one(), BN_value_one(), zero, ctx)) {
-        fprintf(stderr, "BN_mod_exp with zero modulus succeeded!\n");
+    if (BN_mod_exp(a, BN_value_one(), BN_value_one(), zero, ctx))
```
Comment (mattcaswell) [inline test/bntest.c]: TEST_false? etc in this function

File: test/bntest.c [hunk 6]
```diff
@@ -2066,17 +1913,14 @@ static int test_expmodzero()
...
-    if (r == NULL
-            || !BN_generate_prime_ex(r, (int)kBits, 0, NULL, NULL, NULL))
-        goto err;
-    if (BN_num_bits(r) != kBits) {
-        fprintf(stderr, "Expected %u bit prime, got %u bit number\n",
-                kBits, BN_num_bits(r));
+    if (!TEST_ptr(r = BN_new())
+            || !TEST_true(BN_generate_prime_ex(r, (int)kBits, 0,
+                                               NULL, NULL, NULL))
+            || BN_num_bits(r) != kBits)
```
Comment (mattcaswell) [inline test/bntest.c]: TEST_int_eq?

File: test/bntest.c [hunk 7]
```diff
@@ -2208,10 +2044,15 @@ static int file_test_run(STANZA *s)
...
+        if (findattr(s, tp->name) != NULL) {
+            if (!tp->func(s)) {
+                TEST_info("Failed %s test at %d", tp->name, s->start);
+                return 0;
+            }
+            return 1;
+        }
     }
-    fprintf(stderr, "Unknown test at %d\n", s->start);
+    TEST_info("Unknown test at %d", s->start);
```
Comment (mattcaswell) [inline test/bntest.c]: Should this be TEST_error?

File: test/bntest.c [hunk 8]
```diff
@@ -492,17 +473,13 @@ static int test_gf2m_add()
         b->neg = rand_neg();
         BN_GF2m_add(c, a, b);
         /* Test that two added values have the correct parity. */
-        if ((BN_is_odd(a) && BN_is_odd(c))
-            || (!BN_is_odd(a) && !BN_is_odd(c))) {
-            printf("GF(2^m) addition test (a) failed!\n");
+        if (!TEST_false((BN_is_odd(a) && BN_is_odd(c))
+            || (!BN_is_odd(a) && !BN_is_odd(c))))
```
Comment (richsalz) [inline test/bntest.c]: fixed.

File: test/bntest.c [hunk 9]
```diff
@@ -821,17 +796,20 @@ static int test_gf2m_modsqrt()
...
     int p0[] = { 163, 7, 6, 3, 0, -1 };
     int p1[] = { 193, 15, 0, -1 };
 
-    a = BN_new();
-    b[0] = BN_new();
-    b[1] = BN_new();
-    c = BN_new();
-    d = BN_new();
-    e = BN_new();
+
```
Comment (richsalz) [inline test/bntest.c]: fixed.

File: test/bntest.c [hunk 10]
```diff
@@ -928,30 +898,20 @@ static int test_kronecker()
         else if (BN_is_zero(r))
             legendre = 0;
         else {
-            if (!BN_add_word(r, 1))
-                goto err;
-            if (0 != BN_ucmp(r, b)) {
-                printf("Legendre symbol computation failed\n");
```
Comment (richsalz) [inline test/bntest.c]: okya.  broke it into two iffs and added the info call.

File: test/bntest.c [hunk 11]
```diff
@@ -1497,19 +1450,19 @@ static int file_exp(STANZA *s)
...
+            || !TEST_ptr(ret2 = BN_new()))
         goto err;
 
     /* There are two possible answers. */
-    if (!BN_mod_sqrt(ret, a, p, ctx) || !BN_sub(ret2, p, ret))
+    if (!TEST_true(BN_mod_sqrt(ret, a, p, ctx))
+            || !TEST_true(BN_sub(ret2, p, ret)))
         goto err;
 
     if (BN_cmp(ret2, mod_sqrt) != 0
```
Comment (richsalz) [inline test/bntest.c]: oops, missed one!  fixed.

File: test/bntest.c (hunk 1 above)
Comment (richsalz) [inline test/bntest.c]: all fixed, thanks.

File: test/bntest.c (hunk 3 above)
Comment (richsalz) [inline test/bntest.c]: yeah, !TEST_false.  fixed all.

File: test/bntest.c (hunk 2 above)
Comment (richsalz) [inline test/bntest.c]: fixed.

File: test/bntest.c (hunk 7 above)
Comment (richsalz) [inline test/bntest.c]: the caller will report a test failure, so the TEST_info calls in this routine just add more info.

File: test/bntest.c (hunk 6 above)
Comment (richsalz) [inline test/bntest.c]: fixed.

**Comment by richsalz** [comment]
all issues addressed (one is not; added comment at line 1460 to explain why)  new commit pushed, will squash on approval

File: test/bntest.c [hunk 12]
```diff
@@ -1950,78 +1807,59 @@ static int test_negzero()
...
-        fprintf(stderr, "BN_mod_exp with zero modulus succeeded!\n");
+    if (!TEST_false(BN_mod_exp(a, BN_value_one(), BN_value_one(), zero, ctx)))
         goto err;
-    }
     ERR_clear_error();
 
-    if (BN_mod_exp_mont(a, BN_value_one(), BN_value_one(), zero, ctx, NULL)) {
-        fprintf(stderr, "BN_mod_exp_mont with zero modulus succeeded!\n");
+    if (!TEST_false(BN_mod_exp_mont(a, BN_value_one(), BN_value_one(),
+                                   zero, ctx, NULL)))
```
Comment (mattcaswell) [inline test/bntest.c]: off-by-one on the indentation

Comment (richsalz) [inline test/bntest.c]: keen eyes :)  fixed and merging, thanks!

**Pull request closed without merging**